src/main/steps/enter-your-address/get.ts

```diff
@@ -0,0 +1,21 @@
+import autobind from 'autobind-decorator';
+import { Response } from 'express';
+
+import { AppRequest } from '../../app/controller/AppRequest';
+import { GetController } from '../../app/controller/GetController';
+
+import { generateContent } from './content';
+
+@autobind
+export default class YourAddressGetController extends GetController {
+  constructor() {
+    super(__dirname + '/template', generateContent);
+  }
+
+  public async get(req: AppRequest, res: Response): Promise<void> {
+    if (Object.keys(req.query).includes('postcode')) {
+      req.session.userCase.applicant1AddressPostcode = req.query.postcode?.toString();
```

Comment:
is there any way to avoid this? otherwise we'll need a custom controller for each time we ask for an address

Reply:
👍 I've removed these additional get controllers and I hiden the postcode lookup if JavaScript is disabled